File: java/cars-assemble/src/main/java/CarsAssemble.java
import java.lang.Math;

public class CarsAssemble {
    private final static int SPEED_OFF = 0;
    private final static int SPEED_MAXIMUM = 0;
    private final static int CARS_PER_HOUR = 221;

    public static double productionRatePerHour(int speed) {
        double successRate;
        if (speed < 5) {
            successRate = 1.0;
        } else if (speed <= 8) {
            successRate = 0.9;
        } else if (speed == 9) {
            successRate = 0.8;
        } else {
            successRate = 0.77;
        }

        return (double)speed * (double)CARS_PER_HOUR * successRate;
    }

    public static int workingItemsPerMinute(int speed) {
        return (int) Math.floor(CarsAssemble.productionRatePerHour(speed) / 60.0);
    }
}

File: julia/cars-assemble/cars-assemble.jl
function success_rate(speed)
   iszero(speed) && return zero(speed)
   1 ≤ speed ≤ 4 && return 1
   5 ≤ speed ≤ 8 && return 0.9
   speed == 9 ? 0.8 : 0.77
end

production_rate_per_hour(speed) =
    221speed * success_rate(speed)

working_items_per_minute(speed) =
    Int(production_rate_per_hour(speed) ÷ 60)
